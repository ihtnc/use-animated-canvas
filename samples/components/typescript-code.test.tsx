import { describe, test, expect } from 'vitest'
import { render } from '@testing-library/react'
import TypeScriptCode from './typescript-code'

describe('TypeScriptCode component', () => {
  test('should render code with typescript styles', () => {
    const code = 'const message = "Hello, world!";'
    const { container } = render(<TypeScriptCode code={code} />)

    expect(container).toMatchSnapshot()
  })

  test('should apply hljs-specific styles', () => {
    const code = 'const message = "Hello, world!";'
    const { container } = render(<TypeScriptCode code={code} />)

    const keywordValue = container.querySelector('.hljs-keyword')
    const stringValue = container.querySelector('.hljs-string')
    expect(keywordValue).not.toBeNull()
    expect(stringValue).not.toBeNull()
  })
})