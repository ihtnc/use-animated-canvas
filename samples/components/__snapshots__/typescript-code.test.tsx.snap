// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TypeScriptCode component > should render code with typescript styles 1`] = `
<div>
  <span
    class="pb-4"
    data-headlessui-state="open"
    data-open=""
  >
    <h3
      aria-controls="headlessui-disclosure-panel-:r1:"
      aria-expanded="true"
      class="flex group text-xl font-semibold cursor-pointer"
      data-headlessui-state="open"
      data-open=""
      id="headlessui-disclosure-button-:r0:"
    >
      CodeÂ 
      <span
        class="inline-block transition-transform motion-reduce:transform-none
          group-data-[open]:-rotate-90
          group-data-[open]:pl-1
          group-data-[open]:pt-1
          group-hover:group-data-[open]:-translate-y-1
          group-hover:translate-y-1
          rotate-90
          pb-1"
      >
        -&gt;
      </span>
    </h3>
    <div
      data-headlessui-state="open"
      data-open=""
      id="headlessui-disclosure-panel-:r1:"
    >
      <pre>
        <code>
          <span
            class="hljs-keyword"
          >
            const
          </span>
           message = 
          <span
            class="hljs-string"
          >
            "Hello, world!"
          </span>
          ;
        </code>
      </pre>
    </div>
  </span>
</div>
`;
